src/app/services/token-lifetime.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Inject, Injectable } from '@angular/core';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { TranslateService } from '@ngx-translate/core';
+import { WINDOW } from 'app/helpers/window.helper';
+import { Timeout } from 'app/interfaces/timeout.interface';
+import { WebSocketService } from 'app/services';
+import { DialogService } from 'app/services/dialog.service';
+
+@UntilDestroy()
+@Injectable({
+  providedIn: 'root',
+})
+export class TokenLifetimeService {
+  protected timeoutToken: Timeout;
+  protected timeoutLogout: Timeout;
+
+  constructor(
+    private dialogService: DialogService,
+    private translate: TranslateService,
+    private ws: WebSocketService,
+    @Inject(WINDOW) private window: Window,
+  ) {}
+
+  start(addListeners = false): void {
+    if (addListeners) {
+      this.addListeners();
+    }
+
+    this.stop();
+    const lifetime = Number(this.window.localStorage.getItem('lifetime'));
+    this.timeoutToken = setTimeout(() => {
+      this.timeoutLogout = setTimeout(() => this.ws.logout(), 5000);
+      this.dialogService.confirm({
+        title: this.translate.instant('Logout'),
+        message: this.translate.instant('Are you sure you want to logout?'),
```

Comment:
I think the messages should be something like this:
> It looks like your session has been inactive for more than x minutes.
> For security reasons we will log you out at <time>.

or 
> For security reasons we will log you out in <active countdown of x> seconds.

And one button: `Extend session`, which resets the whole thing.